CodeEdit/Features/Settings/Pages/TextEditingSettings/Models/TextEditingSettings.swift

```diff
@@ -175,14 +179,66 @@ extension SettingsData {
             let container = try decoder.container(keyedBy: CodingKeys.self)
             self.size = try container.decodeIfPresent(Double.self, forKey: .size) ?? size
             self.name = try container.decodeIfPresent(String.self, forKey: .name) ?? name
+            self.weight = try container.decodeIfPresent(NSFont.Weight.self, forKey: .weight) ?? weight
         }
 
         /// Returns an NSFont representation of the current configuration.
         ///
         /// Returns the custom font, if enabled and able to be instantiated.
         /// Otherwise returns a default system font monospaced.
         var current: NSFont {
-            return NSFont(name: name, size: size) ?? NSFont.monospacedSystemFont(ofSize: size, weight: .medium)
+            let customFont = NSFont(name: name, size: size)?.withWeight(weight: weight)
+            return customFont ?? NSFont.monospacedSystemFont(ofSize: size, weight: .medium)
+        }
+    }
+}
+
+extension NSFont {
```

Comment:
Could you move these extensions to a separate file? Maybe call it `NSFont+WithWeight.swift`

Reply:
I was going to do that but I’m not sure where we should put this file. Any ideas?

Reply:
Either in the same folder, which might be misleading, so I suggest creating a new directory: `/TextEditingSettings/Extensions/`

Reply:
Done